compiler/damlc/lib/DA/Cli/Damlc.hs

```diff
@@ -518,18 +520,27 @@ withPackageConfig projectPath f = do
 execInit :: Options -> ProjectOpts -> Command
 execInit opts projectOpts =
   Command Init (Just projectOpts) effect
-  where effect = withProjectRoot' projectOpts $ \_relativize ->
-          initPackageDb
+  where effect = withProjectRoot' projectOpts $ \_relativize -> do
+          _ <- initPackageDb
             opts
             (InitPkgDb True)
+          return ()
 
-initPackageDb :: Options -> InitPkgDb -> IO ()
+initPackageDb :: Options -> InitPkgDb -> IO [UnitId]
 initPackageDb opts (InitPkgDb shouldInit) =
-    when shouldInit $ do
+    if shouldInit
+    then do
         isProject <- doesFileExist projectConfigName
-        when isProject $ do
+        if isProject then do
             withPackageConfig defaultProjectPath $ \PackageConfigFields {..} ->
                 createProjectPackageDb opts pSdkVersion pDependencies pDataDependencies
+        else
+           return []
+    else
+      return []
```

Comment:
This breaks `--init-package-db=no`. We need to get the unit ids of the dependencies even if we are not reinitializing the package db. So the flag should probably be passed on to `createProjectPackageDb` which then always returns the unit ids but skips actually overwriting the package db. Given that we haven’t really advertised `--init-package-db=no` and #4413 should make it redundant, we might be able to just break it for now and handle it in a separate PR.

Reply:
this issue still needs to be resolved...

Reply:
The changes were removed from `Damlc.hs`, so nothing to do here